mne/channels/channels.py

```diff
@@ -1536,3 +1539,150 @@ def make_1020_channel_selections(info, midline="z"):
+
+    Parameters
+    ----------
+    instance : instance of Raw, Epochs, or Evoked
+        An MNE-Python object to combine the channels for. The object can be of
+        type Raw, Epochs, or Evoked.
+    groups : dict
+        Specifies which channels are aggregated into a single channel, with
+        aggregation method determined by the ``method`` parameter. One new
+        pseudo-channel is made per dict entry; the dict values must be lists of
+        picks (integer indices of ``ch_names``). For example::
+
+            groups=dict(Left=[1, 2, 3, 4], Right=[5, 6, 7, 8])
+
+        Note that within a dict entry all channels must have the same type.
+    method : str | callable
+        Which method to use to combine channels. If a :class:`str`, must be one
+        of 'mean', 'median', or 'std' (standard deviation). If callable, the
+        callable must accept one positional input (data of shape ``(n_channels,
+        n_times)``, ``(n_epochs, n_channels, n_times)``, or ``(n_evokeds,
+        n_channels, n_time)``) and return an :class:`array <numpy.ndarray>` of
+        shape ``(n_times)``, ``(n_epochs, n_times)``, or ``(n_evokeds,
+        n_times)``. For example with an instance of Raw::
+
+            method = lambda data: np.median(data, axis=0)
+
+        Defaults to ``mean``.
+    keep_stim : bool
+        If True (default False), keep stimulus channels.
+    drop_bad : bool
+        If True (default False), drop channels marked as bad.
+
+    Returns
+    -------
+    combined_inst : instance of Raw, Epochs, or Evoked
+        An MNE-Python object with channels combined for each group defined in
+        the ``groups`` parameter.
+    """
+    from ..io import BaseRaw, RawArray
+    from ..epochs import BaseEpochs
+    from ..evoked import Evoked
+    from .. import EpochsArray, EvokedArray
+
+    # Check to see if data is preloaded
+    _check_preload(instance, 'Combining channels')
+
+    ch_axis = len(instance._data.shape) - 2  # idx of the channel axis
+    ch_idx = list(range(instance.info['nchan']))
+    ch_types = _get_channel_types(instance.info)
+    groups = OrderedDict(deepcopy(groups))
+
+    # Convert string values of ``method`` into callables
+    # XXX Possibly de-duplicate with _make_combine_callable of mne/viz/utils.py
+    if isinstance(method, str):
+        method_dict = {key: partial(getattr(np, key), axis=ch_axis)
+                       for key in ('mean', 'median', 'std')}
+        try:
+            method = method_dict[method]
+        except KeyError:
+            raise ValueError('"method" must be a callable, or one of '
+                             f'"mean", "median", or "std"; got "{method}".')
+
+    # Create new instance
+    if not isinstance(keep_stim, bool):
+        raise TypeError('"keep_stim" must be of type bool, not '
+                        f'{type(keep_stim)}.')
+    if keep_stim is True:
+        try:
+            combined_inst = instance.copy().pick_types(meg=False, stim=True)
```

Comment:
this is bad practice. I you have raw of size 2GB you allocated another 2GB of RAM to get only one channel out. Use mne.pick_types and raw[] syntax. It will be way more efficient.